# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxMacFont.cpp
# Commit: 4c3cb20a5203
# Full Hash: 4c3cb20a52034b3cead6ac88e008f17bd446fb1c
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-08-16 16:01:29
# Regressor Bug: 1740530
# File Overlap Count: 1
# Description:
#   Bug 1740530 - patch 7 - Route COLRv1 fonts on macOS through our COLR-rendering path; only COLRv0 can use the native Core Text support. r=gfx-reviewers,lsalzman
#   
#   Depends on D153880
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152162
# ==============================================================================

diff -r 275eb6c4900b -r 4c3cb20a5203 gfx/thebes/gfxMacFont.cpp
--- a/gfx/thebes/gfxMacFont.cpp	Tue Aug 16 10:39:52 2022 +0000
+++ b/gfx/thebes/gfxMacFont.cpp	Tue Aug 16 10:39:52 2022 +0000
@@ -612,6 +612,16 @@
                        CAIRO_SURFACE_TYPE_QUARTZ;
 }
 
+bool gfxMacFont::UseNativeColrFontSupport() const {
+  if (nsCocoaFeatures::OnHighSierraOrLater()) {
+    auto* colr = GetFontEntry()->GetCOLR();
+    if (colr && COLRFonts::GetColrTableVersion(colr) == 0) {
+      return true;
+    }
+  }
+  return false;
+}
+
 void gfxMacFont::AddSizeOfExcludingThis(MallocSizeOf aMallocSizeOf,
                                         FontCacheSizes* aSizes) const {
   gfxFont::AddSizeOfExcludingThis(aMallocSizeOf, aSizes);